ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/block/BlockManager.java

```diff
@@ -91,6 +98,20 @@ public static BlockManager create(
     return doImportBlock(block);
   }
 
+  @SuppressWarnings("FutureReturnValueIgnored")
+  public SafeFuture<InternalValidationResult> validateAndImportBlock(
+      final SignedBeaconBlock block) {
+    SafeFuture<InternalValidationResult> validationResult = validator.validate(block);
+    validationResult.thenAccept(
+        result -> {
+          if (result.equals(InternalValidationResult.ACCEPT)
+              || result.equals(InternalValidationResult.SAVE_FOR_FUTURE)) {
+            importBlock(block).finish(err -> LOG.error("Failed to process received block.", err));
```

Comment:
Same as for attestations - if it's save for future should we just add it to the pending pool without further processing?

Reply:
Made an issue here to solve later on: https://github.com/ConsenSys/teku/issues/3114